src/vsock/src/main.rs

```diff
@@ -0,0 +1,124 @@
+#[macro_use(crate_version, crate_authors)]
+extern crate clap;
+mod packet;
+mod rxops;
+mod rxqueue;
+mod thread_backend;
+mod txbuf;
+mod vhost_user_vsock_thread;
+mod vhu_vsock;
+mod vsock_conn;
+
+use clap::{App, Arg};
+use std::{
+    process,
+    sync::{Arc, RwLock},
+};
+use vhost::{vhost_user, vhost_user::Listener};
+use vhost_user_backend::VhostUserDaemon;
+use vhu_vsock::{VhostUserVsockBackend, VsockConfig};
+use vm_memory::{GuestMemoryAtomic, GuestMemoryMmap};
+
+/// This is the public API through which an external program starts the
+/// vhost-user-vsock backend server.
+pub fn start_backend_server(vsock_config: VsockConfig) {
+    loop {
```

Comment:
Not sure about this, I think we wanted to have one device per VM as we wanted to integrate it with kata initially. Maybe @stefano-garzarella can correct me?

Reply:
Yep, we support a single VM, also for the configuration would become complicated since we have to provide cid and Unix socket paths for each VM.
Maybe in the future we can extend it, but for now I think is fine to support a single VM, especially for the Kata use case.

Reply:
Okay, no problem. That should be fine for now.